### ISSUE TITLE
spatial/r3: add eigenvalue calculation for Mat type

### ISSUE THREAD

**Issue by soypat** [gonum/gonum#1807]
For symmetric 3x3 matrices:
https://dl.acm.org/doi/abs/10.1145/355578.366316

Singular Value decomp:
https://minds.wisconsin.edu/handle/1793/60736

I' be happy to contribute this feature.

**Comment by calebbuffa** [comment]
Agreed, this would be very helpful as I am currently creating my own solution.

**Comment by vladimir-ch** [comment]
We already have `mat.EigenSym`, `mat.Eigen` and `mat.SVD` for computing these factorizations but I understand the appeal of having optimized versions for 3x3 matrices, so in principle I'm not against this request.

Some random thoughts:

How would the API look like? Would it follow what we have in `mat`?

In `mat`, we have some type safety to do symmetric eigendecomposition only for symmetric matrices. Would we need to introduce a `SymMat` in `r3` or would there be a runtime check or something else?

Would the eigendecomposition compute also eigenvectors or only eigenvalues?

`mat.SVD` gives _non-negative_ singular values while the method from the second reference may return the smallest singular value as negative (to allow U and V be pure rotations). Are we fine with this inconsistency?

How robust are these formulas/algorithms to weird/bad matrices or do they trade it for performance? Good tests would be needed as part of the implementation.
